pkg/experiment/metastore/metastore.go

```diff
@@ -133,11 +140,12 @@ type Metastore struct {
 	readySince time.Time
 
 	dnsProvider *dns.Provider
+	dlq         *dlq.Recovery
 }
 
 type Limits interface{}
 
-func New(config Config, limits Limits, logger log.Logger, reg prometheus.Registerer, client *metastoreclient.Client) (*Metastore, error) {
+func New(config Config, limits Limits, logger log.Logger, reg prometheus.Registerer, client *metastoreclient.Client, bucket objstore.Bucket) (*Metastore, error) {
```

Comment:
I'm almost 100% sure we need to update dependencies for metastore

Reply:
thanks, added `Storage` depdendency